firmware/src/main.rs

```diff
@@ -67,55 +73,74 @@ fn main() -> ! {
 }
 
 #[task]
-async fn control_task(packets: &'static Packets, quat: &'static Unreliable<Quat>) {
+async fn control_task(packets: &'static Packets, quat: &'static Unreliable<Quat>) -> ! {
 	debug!("Control task!");
 
-	loop {
-		match packets.clientbound.recv().await {
-			// Identify ourself when discovery packet is received
-			CbPacket::Discovery => {
-				packets
-					.serverbound
-					.send(SbPacket::Handshake {
-						board: 4,
-						imu: 8,
-						mcu_type: 2,
-						imu_info: (1, 2, 3),
-						build: 5,
-						firmware: "SlimeVR-Rust".into(),
-						mac_address: [0; 6],
-					})
-					.await;
-			}
-			// When heartbeat is received, we should reply with heartbeat 0 aka Discovery
-			// The protocol is asymmetric so its a bit unintuitive. TODO: Split packet type into Server2Client and C2S
-			CbPacket::Heartbeat => {
-				packets.serverbound.send(SbPacket::Heartbeat).await;
-			}
-			// Pings are basically like heartbeats, just echo data back
-			CbPacket::Ping { id } => {
-				packets.serverbound.send(SbPacket::Ping { id }).await;
-			}
+	let sensor_info_fut = async {
+		let mut ticker = Ticker::every(Duration::from_millis(1000));
+		loop {
+			packets
+				.serverbound
+				.send(SbPacket::SensorInfo {
+					sensor_id: 0,     // First sensor (of two)
+					sensor_status: 1, // OK
+					sensor_type: IMU_KIND.protocol_id(),
+				})
+				.await;
+			debug!("Sent SensorInfo");
+
+			ticker.next().await;
+		}
+	};
+	let loop_fut = async {
+		loop {
+			do_work(packets, quat).await;
 		}
+	};
+
+	let (_, _): (Infallible, Infallible) = join(sensor_info_fut, loop_fut).await;
+	unreachable!()
+}
 
-		packets
-			.serverbound
-			.send(SbPacket::SensorInfo {
-				sensor_id: 0,
-				sensor_status: 1,
-				sensor_type: 0,
-			})
-			.await;
-		packets
-			.serverbound
-			.send(SbPacket::RotationData {
-				sensor_id: 0,
-				data_type: 1,
-				quat: quat.wait().await.into_inner().into(),
-				calibration_info: 0,
-			})
-			.await;
+async fn do_work(packets: &Packets, quat: &Unreliable<Quat>) {
+	match packets.clientbound.recv().await {
+		// Identify ourself when discovery packet is received
+		CbPacket::Discovery => {
```

Comment:
if you're connected already, don't echo back

Reply:
how are we connected already?

Reply:
oh, nvm

Reply:
What happens if we don't know if we disconnnected or not? lets say its a new server

Reply:
just check when the last heartbeat was that you've received + some offset

Reply:
will do later